rclpy/test/test_node.py

```diff
@@ -497,26 +504,33 @@ def tearDown(self):
         rclpy.shutdown(context=self.context)
 
     def test_declare_parameter(self):
+        with pytest.raises(ValueError):
+            result_initial_foo = self.node.declare_parameter(
+                'initial_foo', ParameterValue(), ParameterDescriptor())
+        dynamic_typing_descriptor = ParameterDescriptor()
+        dynamic_typing_descriptor.dynamic_typing = True
         result_initial_foo = self.node.declare_parameter(
-            'initial_foo', ParameterValue(), ParameterDescriptor())
+                'initial_foo', ParameterValue(), dynamic_typing_descriptor)
         result_initial_bar = self.node.declare_parameter(
             'initial_bar', 'ignoring_override', ParameterDescriptor(), ignore_override=True)
         result_initial_fizz = self.node.declare_parameter(
             'initial_fizz', 'default', ParameterDescriptor())
         result_initial_baz = self.node.declare_parameter(
-            'initial_baz', ParameterValue(), ParameterDescriptor())
+            'initial_baz', 0., ParameterDescriptor())
         result_initial_buzz = self.node.declare_parameter(
-            'initial_buzz', ParameterValue(), ParameterDescriptor())
+            'initial_buzz', 0., ParameterDescriptor())
         result_initial_foobar = self.node.declare_parameter(
-            'initial_foobar', ParameterValue(), ParameterDescriptor())
+            'initial_foobar', True, ParameterDescriptor())
 
         result_foo = self.node.declare_parameter(
             'foo', 42, ParameterDescriptor())
         result_bar = self.node.declare_parameter(
             'bar', 'hello', ParameterDescriptor())
         result_baz = self.node.declare_parameter(
             'baz', 2.41, ParameterDescriptor())
-        result_value_not_set = self.node.declare_parameter('value_not_set')
+        with warnings.catch_warnings():
+            warnings.filterwarnings('ignore', category=DeprecationWarning)
+            result_value_not_set = self.node.declare_parameter('value_not_set')
```

Comment:
We should also assert that the deprecation message appears, [for example](https://github.com/ros2/rosidl_runtime_py/blob/471ed56db8417b8d16b579383d121a7a6b050d00/test/rosidl_runtime_py/test_import_message.py#L39-L40).

Reply:
Done in https://github.com/ros2/rclpy/pull/683/commits/1e6e49d01f86eb9dd17d0a0d7ba383a3b8e42536.